crossdock/client/ctxpropagation/behavior.go

```diff
@@ -292,6 +294,9 @@ func buildDispatcher(t crossdock.T) (dispatcher yarpc.Dispatcher, tconfig server
 		tconfig.TChannel = &server.TChannelTransport{Host: self, Port: 8087}
 	case "tchannel":
 		outbound = tch.NewOutbound(ch, tch.HostPort(fmt.Sprintf("%s:8082", subject)))
+		tconfig.GRPC = &server.GRPCTransport{Host: self, Port: 8090}
+	case "grpc":
+		outbound = grpc.NewOutbound(fmt.Sprintf("%s:8089", subject))
```

Comment:
I don't know how many of those tests go three levels deep. Consider increasing
depth or using the other transports randomly rather than deterministically.
That is, instead of always going tchannel->GRPC, pick between that and
tchannel->HTTP at random, and do the same for GRPC->HTTP/TChannel,
HTTP->GRPC/TChannel.